AndroidTest/app/src/main/java/cloudant/com/androidtest/keyprovider/KeyManagerTests.java

```diff
@@ -0,0 +1,201 @@
+package cloudant.com.androidtest.keyprovider;
+
+import android.test.AndroidTestCase;
+
+import com.cloudant.sync.datastore.encryption.DPKException;
+import com.cloudant.sync.datastore.encryption.EncryptionKey;
+import com.cloudant.sync.datastore.encryption.KeyManager;
+import com.cloudant.sync.datastore.encryption.KeyStorage;
+
+import java.util.Arrays;
+
+public class KeyManagerTests extends AndroidTestCase {
+    public void testGenerateDPK() {
+        KeyStorage storage = new KeyStorage(getContext(), ProviderTestUtil.getUniqueIdentifier());
+        KeyManager manager = new KeyManager(storage);
+
+        String password = "passw0rd";
+
+        EncryptionKey dpk = manager.generateAndSaveKeyProtectedByPassword(password);
+        assertNotNull("DPK should not be null", dpk);
+        assertNotNull("byte[] of DPK should not be null", dpk.getKey());
+        assertTrue("byte[] of DPK should not be empty", dpk.getKey().length > 0);
```

Comment:
Changes requested in comment included in https://github.com/cloudant/sync-android/commit/dd276d263bfc9c5ddebb047cecd27230cb074f63